lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1053,7 +1053,9 @@ def test_parameter_versions(self):
                     msg=('%s parameter supported only with api_version>=%s'
                          % (pv['param'], pv['min_version'])))
 
-        for publish_def in self.client.module.params.get('publish', []):
+        for publish_def in params['publish'] or []:
+            if not isinstance(publish_def, dict):
```

Comment:
Note, usually, one should check whether something is an instance of an abstractbaseclass instead of dict, list, or other builtin types.  For instance:
``` python
from ansible.module_utils.common._collections_compat import Mapping
[...]
if not isinstance(publish_def, Mapping):
```

Reply:
Thanks, changed the PR accordingly!